### PR TITLE
Clean up StringTable code

### PR DESCRIPTION
**Pull request by gacholio**
- inline various functions that were only ever used here (and should
never be used anywhere else)
- remove J9_STR_ASCII which was added solely to communicate to one of
the now-inlined functions
- replace repeated bit tests with boolean variables

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

File: runtime/gc_base/StringTable.cpp [hunk 1]
```diff
@@ -532,21 +532,35 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
...
+	Trc_MM_createJavaLangString_Entry(vmThread, length, data, stringFlags);
+
+	/* Determine if the string contains only ASCII (<= 0x7F) characters */
+	bool isASCII = true;
 	if (isUnicode) {
-		if (IS_STRING_COMPRESSION_ENABLED_VM(vm) && VM_VMHelpers::isUnicodeASCII((const U_16 *)data, length / 2)) {
-			stringFlags |= J9_STR_ASCII;
+		/* Translation and anon class name are not currently supported for unicode input */
+		Assert_MM_false(translateSlashes || anonClassName);
+		U_16 *unicodeData = (U_16*)data;
```
Comment (pshipton) [inline runtime/gc_base/StringTable.cpp:548]: This discards the optimization of only needing to determine isASCII when compressStrings is true.

Comment (pshipton) [reply runtime/gc_base/StringTable.cpp:548]: nm, I see you are using it below.

Comment (pshipton) [reply runtime/gc_base/StringTable.cpp:548]: Ah, but not in the isUnicode case, so my original comment stands.

Comment (gacholio) [reply runtime/gc_base/StringTable.cpp:548]:
```
		if (isUnicode) {
			U_16 * unicodeData = (U_16 *) data;

			if (compressStrings && isASCII) {
				for (UDATA i = 0; i < unicodeLength; ++i) {
					J9JAVAARRAYOFBYTE_STORE(vmThread, charArray, i, (I_8)unicodeData[i]);
				}
```

Your previous code actually removed this optimization (which was hidden in the VMHelpers code).

Comment (gacholio) [reply runtime/gc_base/StringTable.cpp:548]: Hmm, I might be wrong about the actual impact - looks like isASCII is only used  for unicode if compression is on. Part of the reason for this cleanup is that the logic was so impenetrable.

I'll re-add the optimization with a comment - I suppose the worst thing that could happen with any future changes is that we take the long path for unicode, which would still yield the correct result.

Comment (gacholio) [reply runtime/gc_base/StringTable.cpp]: Fixed.

**Comment by pshipton** [comment]
jenkins test sanity,extended,sanity.openjdk zlinux jdk11

**Comment by pshipton** [comment]
jenkins compile win jdk8

**Comment by pshipton** [comment]
```
01:38:34  StringTable.cpp(244) : error C2220: warning treated as error - no 'object' file generated
01:38:34  StringTable.cpp(244) : warning C4244: '=' : conversion from 'UDATA' to 'U_32', possible loss of data
01:38:34  StringTable.cpp(387) : warning C4244: '=' : conversion from 'UDATA' to 'U_32', possible loss of data
```

**Comment by gacholio** [comment]
Windows errors fixed.

**Comment by pshipton** [comment]
jenkins compile win jdk8

**Review by pshipton — APPROVED** [review]

**Pull request merged**